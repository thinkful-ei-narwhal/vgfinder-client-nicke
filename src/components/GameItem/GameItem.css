.gameitem-display {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

.text-holder > p {
  color: lightgray;
  width: 300px;
}

.textbox {
  margin: 3px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.game-item-container {
  background-color: #333;
  margin-bottom: 20px;
  padding-top: 15px;
  padding-bottom: 15px;
  min-width: 320px;
}

/* Aspect ratio stuff */

.game-item-img-container {
  width: 350px;
  height: 197px;
  margin: 0 auto;
}

.game-item-img {
  width: 100%;
  height: 100%;
}

@media only screen and (max-width: 520px) {
  .textbox {
    margin: 0 auto;
  }

  .game-item-container {
    background-color: #333;
    height: 380px;
    padding-top: 15px;
  }

  .game-item-img-container {
    width: 300px;
    height: 169px;
  }

  .gameitem-display {
    flex-direction: column;
    justify-content: space-between;
    height: 350px;
  }
}
