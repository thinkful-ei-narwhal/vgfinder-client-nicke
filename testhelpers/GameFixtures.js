const GameFixtures = {
  stubGameData: {
    id: "1",
    title: "Hollow Knight",
    description:
      "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
    genre: "Adventure",
    rating: "4",
    developer: "Team Cherry",
    trailer_url:
      "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
    image_url_box_art:
      "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
    image_url_two:
      "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
    image_url_three:
      "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
  },
  stubGameListData: [
    {
      id: "1",
      title: "Hollow Knight",
      description:
        "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
      genre: "Adventure",
      rating: "4",
      developer: "Team Cherry",
      trailer_url:
        "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
      image_url_box_art:
        "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
      image_url_two:
        "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
      image_url_three:
        "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
    },
    {
      id: "2",
      title: "Hollow Knight",
      description:
        "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
      genre: "Adventure",
      rating: "4",
      developer: "Team Cherry",
      trailer_url:
        "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
      image_url_box_art:
        "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
      image_url_two:
        "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
      image_url_three:
        "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
    },
  ],
};

export default GameFixtures;
