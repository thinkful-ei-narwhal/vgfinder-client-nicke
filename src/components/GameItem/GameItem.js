import './GameItem.css'
import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import UserIdContext from '../../contexts/UserIdContext';
import TokenService from './../../services/token-service';
import AddToWishlist from '../AddToWishlist/AddToWishlist'
import RemoveFromWishlistButton from '../RemoveFromWishlistButton/RemoveFromWishlistButton';

export default class GameItem extends Component {
  static contextType = UserIdContext;

  generateWishlistButton(wishlist, game) {
    if (TokenService.hasAuthToken()) {
      return !!wishlist.find(wishlistItem => wishlistItem.game_id === game.id)
        ? <RemoveFromWishlistButton game={game} />
        : <AddToWishlist game={game} />;
    }
    else {
      return <p>Sign in to add game to wishlist!</p>;
    }
  }

  render() {
    const wishlist = this.context.wishlist;
    return (
      <div>
        <Link to={`/games/${this.props.game.id}`}><img src={this.props.game.image_url_box_art} width="300" alt="Logo" /></Link>
        <br />
        {this.generateWishlistButton(wishlist, this.props.game)}
        <p>
          Title: {this.props.game.title}
        </p>
        <p>
          Genre: {this.props.game.genre}
        </p>
        <p>
          Rating: {this.props.game.rating}
        </p>
        <p>
          Developer: {this.props.game.developer}
        </p>
      </div>
    )
  }
}