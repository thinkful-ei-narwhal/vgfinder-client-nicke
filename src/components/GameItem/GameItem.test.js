import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import GameItem from "./GameItem";
import GameFixtures from "./../../../testhelpers/GameFixtures";

describe(`GameItem component`, () => {
  it("renders a GameItem with supplied props", () => {
    const wrapper = shallow(<GameItem game={GameFixtures.stubGameData} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
