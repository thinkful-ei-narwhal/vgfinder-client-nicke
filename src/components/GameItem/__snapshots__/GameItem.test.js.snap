// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameItem component renders a GameItem with supplied props 1`] = `
<div
  className="game-item-container"
>
  <div
    className="gameitem-display"
  >
    <div
      className="game-item-img-container"
    >
      <Link
        to="/games/1"
      >
        <div
          className="container"
          style={
            Object {
              "backgroundColor": "black",
              "backgroundImage": "url(https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500)",
              "backgroundPosition": "center",
              "backgroundRepeat": "no-repeat",
              "backgroundSize": "contain",
            }
          }
        />
      </Link>
    </div>
    <div
      className="text-holder"
    >
      <p
        className="textbox"
      >
        Title: 
        Hollow Knight
      </p>
      <p
        className="textbox"
      >
        Rating: 
        4
      </p>
      <p
        className="textbox"
      >
        Genre: 
        Adventure
      </p>
      <p
        className="textbox"
      >
        Developer: 
        Team Cherry
      </p>
      <Button
        disabled={true}
      >
        Sign in to add game to wishlist
      </Button>
    </div>
  </div>
</div>
`;
